@here/harp-mapview/lib/text/TextElementStateCache.ts

```diff
@@ -242,40 +252,57 @@ export class TextElementStateCache {
     }
 
     private findDuplicate(
-        elementState: TextElementState,
+        element: TextElement,
+        cacheKey: string | number,
         zoomLevel: number
     ): { entries: TextElementState[]; index: number } | undefined {
-        const element = elementState.element;
-
         // Point labels may have duplicates (as can path labels), Identify them
         // and keep the one we already display.
 
-        const cachedEntries = this.m_textMap.get(element.text);
+        const cachedEntries = this.m_textMap.get(cacheKey);
 
         if (cachedEntries === undefined) {
-            // No labels found with the same text.
+            // No labels found with the same key.
             return undefined;
         }
 
-        const maxSqDistError = getDedupSqDistTolerance(zoomLevel);
+        tmpCachedDuplicate.entries = cachedEntries;
+
+        if (element.hasFeatureId()) {
+            // Duplicate with same feature id found.
+            assert(cachedEntries.length === 1);
+            const cachedElement = cachedEntries[0].element;
+            assert(element.featureId === cachedElement.featureId);
+
+            if (cachedElement.text === element.text) {
+                tmpCachedDuplicate.index = 0;
+            } else {
+                tmpCachedDuplicate.index = -1;
+                logger.debug(
+                    `Text feature id ${element.featureId} collision between "${element.text} and \
+                     ${cachedElement.text}`
```

Comment:
if someone sees this message, what should they do? Should they create a ticket with some specific team to let them know that there is an id collision? Or can it be safely ignored?

Reply:
This would happen if there's some issue on map compilation side and they don't assign unique feature ids. It should be investigated.